client.go

```diff
@@ -81,15 +84,29 @@ func NewClient(ctx *Context, cfg *Config, w *Wallet) (*Client, error) {
 	funder := ethchannel.NewETHFunder(cb, common.Address(cfg.AssetHolder.addr))
 	c := client.New(acc, dialer, funder, adjudicator, w.w)
 
+	// Set up persistence
+	db, err := leveldb.LoadDatabase(cfg.DatabasePath)
+	if err != nil {
+		return nil, errors.WithMessage(err, "creating/loading database")
+	}
+	persister, err := keyvalue.NewPersistRestorer(db)
+	if err != nil {
+		return nil, errors.WithMessage(err, "creating PersistRestorer")
+	}
+
+	c.EnablePersistence(persister)
```

Comment:
Now looking at this, we should also make it possible to make persistence optional.
I think the best way forward would be to
- remove the `DatabasePath` from the `Config` struct,
- add method `Client.EnablePersistence(dbPath string)` within which you do this whole persistence setup (and set `c.persister`),
- in `Client.Close` you only call `c.persister.Close()` if `c.persister != nil`.
- in `Client.Reconnect` you return an error saying "persistence not enabled"  if `c.persister == nil`.

Reply:
Well, then we can also remove the `Reconnect` function and do it in the `EnablePersistence`, or?

Reply:
I think it is better to give the user control when and whether to start the reconnect. It might still happen that you want to enable persistence and listen for incoming sync updates, but not initiate the reconnect yourself. So I think it is better to keep this separate.

In the Java example, please do both after one another, `enablePersistence(dbPath)` followed by `reconnect()` with a timeout like 10 sec.